cpp/src/indexing/construction/detail/phase_1.cuh

```diff
@@ -57,11 +58,11 @@ compute_point_keys_and_sorted_indices(cudf::column_view const &x,
                                       T y_max,
                                       T scale,
                                       int8_t max_depth,
-                                      rmm::mr::device_memory_resource *mr,
-                                      cudaStream_t stream)
+                                      rmm::cuda_stream_view stream,
+                                      rmm::mr::device_memory_resource *mr)
 {
   rmm::device_uvector<uint32_t> keys(x.size(), stream);
-  thrust::transform(rmm::exec_policy(stream)->on(stream),
+  thrust::transform(rmm::exec_policy(stream)->on(stream.value()),
```

Comment:
Unrelated to this PR, but it'd be neat if `rmm::cuda_stream_view` could produce the `rmm::exec_policy` for us like this:
```suggestion
  thrust::transform(stream.exec_policy(),
```

Reply:
Neat idea. We are revamping exec_policy, should consider this...

Reply:
`cuda_stream_view` shouldn't need to know anything about Thrust or execution policies. With the planned exec policy simplifications, this will be simplified to just:
```
thrust::transform(rmm::exec_policy(stream),
```

Reply:
Yeah, I guess it doesn't save anything to do it the other way around.